From 7953867ba46223dfab373ce58cff0a4bc20c3864 Mon Sep 17 00:00:00 2001
From: Felix Geyer <debfx@ubuntu.com>
Date: Tue, 12 May 2015 13:45:43 -0400
Subject: [PATCH] FindJava: Fix OpenJDK 8 version detection (#15565)

The openjdk-8 (8u45-b14-2) package on Debian unstable has extra
text after the version number components.  Match this and add it
to the reported version string.
---
 Modules/FindJava.cmake |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/FindJava.cmake b/Modules/FindJava.cmake
index bb73853..9e43174 100644
--- a/Modules/FindJava.cmake
+++ b/Modules/FindJava.cmake
@@ -137,7 +137,7 @@ if(Java_JAVA_EXECUTABLE)
       elseif(var MATCHES "java full version \"kaffe-([0-9]+\\.[0-9]+\\.[0-9_]+)\"")
         # Kaffe style
         set(Java_VERSION_STRING "${CMAKE_MATCH_1}")
-      elseif(var MATCHES "openjdk version \"([0-9]+\\.[0-9]+\\.[0-9_]+)\"")
+      elseif(var MATCHES "openjdk version \"([0-9]+\\.[0-9]+\\.[0-9_]+.*)\"")
         # OpenJDK ver 1.7.x on OpenBSD
         set(Java_VERSION_STRING "${CMAKE_MATCH_1}")
       else()
-- 
1.7.10.4

